File: apps/admin_mobile/lib/domain/repositories/tenant_repository.dart
abstract class TenantRepository {
  Future<Map<String, dynamic>> getSettings();
  Future<Map<String, dynamic>> updateSettings(Map<String, dynamic> data);
  Future<List<Map<String, dynamic>>> getImages();
  Future<Map<String, dynamic>> addImage(String filePath);
  Future<void> deleteImage(int id);
  Future<void> reorderImages(List<int> ids);
}

File: apps/admin-v2/src/domain/repositories/tenant.repository.ts
import type {
  TenantSettings,
  TenantImage,
  BillingProfile,
  BillingProfileInput,
  SriLookupResult,
  TaxIdType,
} from '../entities/tenant';

export interface TenantRepository {
  getSettings(): Promise<TenantSettings>;
  updateSettings(data: Partial<TenantSettings>): Promise<TenantSettings>;
  getImages(): Promise<TenantImage[]>;
  addImage(file: File): Promise<TenantImage>;
  deleteImage(id: string): Promise<void>;
  reorderImages(ids: string[]): Promise<void>;
  getBillingProfile(): Promise<BillingProfile>;
  updateBillingProfile(input: BillingProfileInput): Promise<BillingProfile>;
  lookupTaxId(type: TaxIdType, taxId: string): Promise<SriLookupResult>;
}
